xonsh/aliases.py

```diff
@@ -406,7 +406,7 @@ def xonsh_reset(args, stdin=None):
 def source_foreign_fn(
     shell: str,
     files_or_code: Annotated[list[str], Arg(nargs="+")],
-    interactive=True,
+    interactive=False,
```

Comment:
if the default becomes False, the argument should be updated as `--interactive` from `--non-interactive` . It is specified in the docstring

Reply:
Ahh, right.  @jnoortheen do you have a pattern with the argparse aliases to keep the argument `-n` around but have it be a no-op?  I realized there are a few downstream libraries that depend on the `-n` flag.